driver/src/contracts/stablex_contract.rs

```diff
@@ -128,9 +130,14 @@ impl<'a> StableXContract for StableXContractImpl<'a> {
                 prices,
                 token_ids_for_price,
             )
-            .call()
-            .wait()?;
+            .view();
 
+        let builder = if let Some(block_number) = block_number {
+            builder.block(block_number)
```

Comment:
Isn't going to run into the same error we had when using the block number for paginated order book reading where the node did not support this properly?

Reply:
I think there the underlying problem was that we were load-balancing between multiple nodes and some nodes might have not yet seen the block. Thus we were actually querying for a future block (as far as that node was concerned). We checked the config and should have been able to query up to 64 blocks in the past.

I guess the view function should be fairly fast to execute so that a race-condition is unlikely (meaning we could just omit the block number). I'd be keen to give it another try and see how it behaves in staging.

What do you think?

Reply:
I didn't realize it should have been possible. If 64 blocks should work then it's definitely fine.